drivers/storage/partmgr/partition.c

```diff
@@ -694,6 +697,38 @@ PartitionHandleDeviceControl(
             Irp->IoStatus.Information = sizeof(*volExts);
             break;
         }
+        case IOCTL_VOLUME_QUERY_VOLUME_NUMBER:
+        {
+            PVOLUME_NUMBER volNum = Irp->AssociatedIrp.SystemBuffer;
+            if (!VerifyIrpOutBufferSize(Irp, sizeof(*volNum)))
+            {
+                status = STATUS_BUFFER_TOO_SMALL;
+                break;
+            }
+
+            PartMgrAcquireLayoutLock(fdoExtension);
+
+            volNum->VolumeNumber = partExt->VolumeNumber;
+            RtlCopyMemory(volNum->VolumeManagerName,
+                          L"VOLMGR  ", // Must be 8 space-padded characters
+                          sizeof(volNum->VolumeManagerName));
+
+            PartMgrReleaseLayoutLock(fdoExtension);
+
+            status = STATUS_SUCCESS;
+            Irp->IoStatus.Information = sizeof(*volNum);
+            break;
+        }
+        case IOCTL_VOLUME_IS_PARTITION:
+        {
+            // The only type of volume we support right now is disk partition
```

Comment:
Nit: I would use a multiline comment here. No big deal either way, though.

Reply:
If it were me, I would, but I tried here to keep consistent with the rest of the file.